### PR TITLE
Move Kotlin DSL script caching tests to `soak`

### PR DESCRIPTION
**Pull request by bamboo**
They are very expensive and don't need to be executed often.

### CODE REVIEW

File: subprojects/soak/soak.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+gradlebuildJava {
+    moduleType = ModuleType.INTERNAL
+}
+
+testFixtures {
+    from(":core")
+}
+
+tasks.matching { it.name in listOf("integTest", "java9IntegTest") }.configureEach {
+    require(this is Test)
```
Comment (eskatos) [inline subprojects/soak/soak.gradle.kts]: :o: could be replaced by chaining `.withType<Test>()` in the above statement

**Comment by lptr** [comment]
The work I'm doing around the execution engine might be relevant here: the execution engine is used now by tasks and transforms to run with caching and other features with little configuration. I'm planning to move Kotlin script compilation over to use the engine once I finished the couple of features remaining.

File: subprojects/soak/soak.gradle.kts (hunk 1 above)
Comment (bamboo) [reply subprojects/soak/soak.gradle.kts]: After a conversation with @jjohannes I've removed the reference to `java9IntegTest` then was able to replace the whole thing by configuration via type-safe accessor.

**Comment by bamboo** [comment]
PR is ready for another review, @eskatos.

**Pull request merged**